charger/easee.go

```diff
@@ -306,6 +320,53 @@ func (c *Easee) ProductUpdate(i json.RawMessage) {
 	case easee.DYNAMIC_CHARGER_CURRENT:
 		c.dynamicChargerCurrent = value.(float64)
 	case easee.CHARGER_OP_MODE:
+		//for relevant op mode changes, leaving op mode 1 and 3, or reaching op mode 1 and 7, request new update ofLIFETIME_ENERGY
+		if !(c.opMode == value.(int)) &&  //only if op mode actually changed AND
+			(easee.ModeDisconnected == c.opMode || easee.ModeCharging == c.opMode || //from these op modes
+			 easee.ModeDisconnected == value.(int) || easee.ModeAwaitingAuthentication == value.(int)) { //or to these op modes
+			c.log.TRACE.Printf("Trigger update of LIFETIME_ENERGY, CHARGER_OP_MODE changed from %v to %v", c.opMode, value.(int));
+			c.RequestLifetimeEnergyUpdate()
+		}
+
+		//OpMode changed from "no car" to some other state 
+		//Assume car connected if easee is not offline (mode 0) or still disconnected (mode 1)
+		//Looks like a new charging session is about to start, so reset internal value of SESSION_ENERGY to 0,
+		//and observation timestamp to "now". This should be done in a proper way by the api, but it's not.
+		if easee.ModeDisconnected == c.opMode && easee.ModeAwaitingStart <= value.(int) {
+			c.log.TRACE.Printf("ProductUpdate: Car connected. Reset internal value of SESSION_ENERGY to 0");
+			c.sessionEnergy = 0;
+			//use -1. Using 0 could not distinguish between reset after session and newly started evcc
+			c.sessionStartLifetimeEnergy = -1;
```

Comment:
nitpick: move -1 into a well named constant. Can get rid of inline comment then.

Edit: That would include using the constant above as well, of course.